macos/FluentUI/Separator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFSeparatorStyle)
+public enum SeparatorStyle: Int {
+    case `default`
+    case shadow
+
+    fileprivate var color: NSColor {
+		switch self {
+		case .default:
+			if #available(OSX 10.14, *) {
+				return NSColor.separatorColor
+
+			} else {
+				return NSColor.systemGray
+			}
+		case .shadow:
+			return NSColor.gridColor
+        }
+    }
+}
+
+@objc(MSFSeparatorOrientation)
+public enum SeparatorOrientation: Int {
+    case horizontal
+    case vertical
+}
+
+
+@objc(MSFSeparator)
+open class Separator: NSView {
+	private var orientation: SeparatorOrientation = .horizontal
+	
+	private var style: SeparatorStyle = .default
+
+	@objc public override init(frame frameRect: NSRect) {
+		super.init(frame: frameRect)
+		initialize()
+	}
+	
+	@objc public init(orientation: SeparatorOrientation = .horizontal) {
+		super.init(frame: .zero)
+		initialize(orientation: orientation)
+	}
+
+	@objc public init(style: SeparatorStyle = .default, orientation: SeparatorOrientation = .horizontal) {
+		super.init(frame: .zero)
+		initialize(style: style, orientation: orientation)
+	}
+
+	@available(*, unavailable)
+	required public init?(coder aDecoder: NSCoder) {
+		preconditionFailure()
+	}
+
+	private func initialize(style: SeparatorStyle = .default, orientation: SeparatorOrientation = .horizontal) {
+		wantsLayer = true
+		self.style = style
+		self.orientation = orientation
+		switch orientation {
+		case .horizontal:
+			heightAnchor.constraint(equalToConstant: 1.0).isActive = true
+		case .vertical:
+			widthAnchor.constraint(equalToConstant: 1.0).isActive = true
+		}
+	}
+	
+	open override var wantsUpdateLayer: Bool {
+		return true
+	}
+	
+	open override func updateLayer() {
+		if let layer = layer {
+			layer.backgroundColor = style.color.cgColor
+		}
+	}
+}
```

Comment:
Added this comment on the commit I removed the ovverride, but ew-commenting here so it's not lost.
iOS contains the following override:
```
	open override var intrinsicContentSize: CGSize {
		switch orientation {
		case .horizontal:
			return CGSize(width: NSView.noIntrinsicMetric, height: frame.height)
		case .vertical:
			return CGSize(width: frame.width, height: NSView.noIntrinsicMetric)
		}
	}
```
 I'm not sure the benefit of specifying NSView.noIntrinsicMetric, so I removed it. Would anyone have insight on what it is useful for? 

I found this within the documentation for AppKit.NSLayoutConstraint

"Note that not all views have an intrinsicContentSize. A horizontal slider has an intrinsic height, but no intrinsic width - the slider artwork has no intrinsic best width. A horizontal NSSlider returns (NSViewNoIntrinsicMetric, ) for intrinsicContentSize. An NSBox returns (NSViewNoIntrinsicMetric, NSViewNoIntrinsicMetric). The intrinsic content size is concerned only with data that is in the view itself, not in other views."

Reply:
In this case we should use intrinsicContentSize instead of a constraint (can remove that code in initialize) - in general the principle of autolayout is that a view knows the size it wants to be if it's not a container of other views. NoIntrinsicMetric means the control is not particular about its size in a particular dimension, which makes sense for a separator. It's particular about its width when vertical or its height when horizontal,  but otherwise requires its container to constrain it to the size it wants